src/server/page_server.cpp

```diff
@@ -205,6 +206,13 @@ page_server::connection_handler::receive_start_catch_up (tran_server_conn_t::seq
 
   er_log_debug (ARG_FILE_LINE, "receive_start_catch_up: hostname = %s, port = %d, LSA = (%lld|%d)\n", host.c_str (), port,
 		LSA_AS_ARGS (&catchup_lsa));
+  if (port == -1)
+    {
+      return; // TODO: It means that the ATS is booting up, it will be set properly after ATS recovery is implemented.
+    }
+
+  // TODO: A thread will take the catch-up including establishing connection to avoid blocking ATS->PS reqeusts.
+  m_ps.connect_to_leader_page_server (std::move (host), port);
```

Comment:
1. Basically, it's a logical naming. More than one node can be leader.
2. We can guarantee that at least one node can be the leader. If there is only one reliable node, several page servers request the catch-up to that node. There is room to improve this with a catch-up policy. For example, 
    - While booting up the ATS, we can delay some catch-up to assign it to another node  which lesser falls behind so is expected get the catch-up done quickly.
    - While ruuing the ATS, there are at least majority of nodes are realiable, so we can spread the catch-up requests. In this case, we can make one node at most is assigned a PS.

Reply:
The keyword 'leader' seems a bit unsuitable to me. Because it is usually thought to contain the meaning of being the only one of many. Accordingly, it can be felt that the PS that serves as the leader role among the PS clusters exists.

Reply:
I agree it can be confusing. I actually struggled to come up with a proper name. Could you recommend more suitable names?

Reply:
How about supporting a general interface to connect to page server like connect_to_page_server() if there is no special action for catch up in the function?

Reply:
How about creating what it's needed and expanding it when more is needed? It's my approach and I believe that it simplifies the implementation. 
For example, I presume that requests for the catch-up are one-way from a follower ps to a leader ps, and the leader ps just reply. With this, we can simplify the disconnection process. When the follower ps wants to disconnect after it's done the catch-up, it can just disconnect.

Reply:
Sorry, I catch your response too late.

I respect your approach. It is very good to me. However, we need to focus on the changes of the naming 'leader'. Do you have any good candidates for it? Or do you just use it? I can see that "leader" has a semantic issue, but I can't come up with a suitable alternative either.

Reply:
It seems better to use "followee" instead of "leader".

Reply:
or "catchup_server" and "catchup_client"?

Reply:
Follower is used. So, "followee" would be good to me.